src/MediatR.Extensions.Autofac.DependencyInjection/Builder/MediatRConfigurationBuilder.cs

```diff
@@ -146,16 +149,73 @@ public MediatRConfigurationBuilder WithCustomStreamPipelineBehaviors(IEnumerable
     {
         this.internalOpenGenericHandlerTypesToRegister.Add(openHandlerType);
     }
+
+    /// <summary>
+    /// Register all open-generic handler-types with a callback. from <see cref="MediatR.Extensions.Autofac.DependencyInjection.KnownHandlerTypes"/>
+    /// </summary>
+    /// <param name="openGenericHandlerTypeToRegisterCallback">Delegate callback</param>
+    /// <returns>Self configured instance</returns>
+    public MediatRConfigurationBuilder WithOpenGenericHandlerTypeToRegisterCallback(Action<IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>> openGenericHandlerTypeToRegisterCallback)
+    {
+        foreach (var type in internalOpenGenericHandlerTypesToRegister)
+        {
+            WithOpenGenericHandlerTypeToRegisterCallback(type, openGenericHandlerTypeToRegisterCallback);
+        }
+        return this;
+    }
+    
+    
+    /// <summary>
+    /// Register a callback for a specific open-generic handler-type.
+    /// </summary>
+    /// <param name="openGenericHandlerTypeToRegisterCallback"></param>
+    /// <typeparam name="TOpenGenericType">Type from <see cref="MediatR.Extensions.Autofac.DependencyInjection.KnownHandlerTypes"/></typeparam>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException">If type that registered is out of scope of <see cref="MediatR.Extensions.Autofac.DependencyInjection.KnownHandlerTypes"/></exception>
+    internal MediatRConfigurationBuilder WithOpenGenericHandlerTypeToRegisterCallback<TOpenGenericType>(Action<IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>> openGenericHandlerTypeToRegisterCallback)
+    {
+        var openGenericType = typeof(TOpenGenericType);
+        
+        return WithOpenGenericHandlerTypeToRegisterCallback(openGenericType, openGenericHandlerTypeToRegisterCallback);
+    }
+    
+    /// <summary>
+    /// Register a callback for a specific open-generic handler-type.
+    /// </summary>
+    /// <param name="openGenericType">Type of Handler like (IRequestHandler, INotificationHandler) etc.</param>
+    /// <param name="openGenericHandlerTypeToRegisterCallback">Callback</param>
+    /// <returns>Self configured instance</returns>
+    /// <exception cref="ArgumentException">If type that registered is out of scope of <see cref="MediatR.Extensions.Autofac.DependencyInjection.KnownHandlerTypes"/></exception>
+    internal MediatRConfigurationBuilder WithOpenGenericHandlerTypeToRegisterCallback(Type openGenericType, Action<IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>> openGenericHandlerTypeToRegisterCallback)
+    {
+        if (!KnownHandlerTypes.AllTypes.Contains(openGenericType))
+        {
+            throw new ArgumentException(
+                $"Invalid open-generic handler-type {openGenericType.Name}",
+                nameof(openGenericType));
+        }
+
+        if (this.internalCustomRegistrationAction.ContainsKey(openGenericType))
```

Comment:
No if / else. Please just return inside the if, otherwise execute the remaining bits. 

You can just amend your commit and force push.